pkg/controller/machine/machine_controller.go

```diff
@@ -623,39 +624,53 @@ func (r *Reconciler) deleteCloudProviderInstance(prov cloudprovidertypes.Provide
 	})
 }
 
-func (r *Reconciler) deleteNodeForMachine(ctx context.Context, machine *clusterv1alpha1.Machine) error {
+func (r *Reconciler) deleteNodeForMachine(ctx context.Context, machine *clusterv1alpha1.Machine) (*reconcile.Result, error) {
+	// List all the volumeAttachments in the cluster; we must be sure that all
+	// of them will be deleted before deleting the node
+	volumeAttachments := &storagev1.VolumeAttachmentList{}
```

Comment:
Can we just abstract this behaviour in a method and only run it if the cloud provider is vSphere.